tools/devops/automation/templates/packages/build.yml

```diff
@@ -0,0 +1,440 @@
+        Write-Host "##vso[task.setvariable variable=BuildNugets;isOutput=true]False"
+      } else {
+        Write-Host "##vso[task.setvariable variable=BuildNugets;isOutput=true]True"
+      }
+
+      if ($Env:SkipSigning -eq "True") {
+        Write-Host "##vso[task.setvariable variable=SignPkgs;isOutput=true]False"
+      } else {
+        Write-Host "##vso[task.setvariable variable=SignPkgs;isOutput=true]True"
+      }
+
+      if ($Env:SkipExternalTests -eq "True") {
+        Write-Host "##vso[task.setvariable variable=RunExternalTests;isOutput=true]False"
+      } else {
+        Write-Host "##vso[task.setvariable variable=RunExternalTests;isOutput=true]True"
+      }
+
+      if ($Env:SkipPackagedXamarinMacTests -eq "True") {
+        Write-Host "##vso[task.setvariable variable=RunMacTests;isOutput=true]False"
+      } else {
+        Write-Host "##vso[task.setvariable variable=RunMacTests;isOutput=true]True"
+      }
+
+      if ($Env:SkipPublicJenkins -eq "True") {
+        Write-Host "##vso[task.setvariable variable=SkipPublicJenkins;isOutput=true]True"
+      } else {
+        Write-Host "##vso[task.setvariable variable=SkipPublicJenkins;isOutput=true]False"
+      }
+
+      Write-Host "##vso[task.setvariable variable=RunSampleTests;isOutput=true]$Env:RunSampleTests"
+      Write-Host "##vso[task.setvariable variable=RunInternalTests;isOutput=true]$Env:RunInternalTests"
+
+    } else {
+      # set the defaults, all the things! o/
+      Write-Host "##vso[task.setvariable variable=IsPR;isOutput=true]False"
+
+      # build pkg, nugets and sign them
+      Write-Host "##vso[task.setvariable variable=BuildPkgs;isOutput=true]True"
+      Write-Host "##vso[task.setvariable variable=BuildNugets;isOutput=true]True"
+      Write-Host "##vso[task.setvariable variable=SignPkgs;isOutput=true]True"
+
+      # tests, run all of them, internal, external, mac but not sample tests
+      Write-Host "##vso[task.setvariable variable=RunInternalTests;isOutput=true]True"
+      Write-Host "##vso[task.setvariable variable=RunExternalTests;isOutput=true]True"
+      Write-Host "##vso[task.setvariable variable=RunMacTests;isOutput=true]True"
+      Write-Host "##vso[task.setvariable variable=RunSampleTests;isOutput=true]False"
+      Write-Host "##vso[task.setvariable variable=SkipPublicJenkins;isOutput=true]False"
+
+      # if a developer decided to trigger one without device tests, allow it
+      if ($Env:RUN_DEVICE_TESTS -eq "true") {
+        Write-Host "##vso[task.setvariable variable=RunDeviceTests;isOutput=true]True"
+      } else {
+        Write-Host "##vso[task.setvariable variable=RunDeviceTests;isOutput=true]False"
+      }
+    }
+
+  name: configuration
+  displayName: "Parse PR labels"
+  timeoutInMinutes: 5
+  env:
+    RUN_DEVICE_TESTS: '${{ parameters.runDeviceTests }}'
+
+- bash: |
+    set -x
+    set -e
+
+    if [[ "$IsPR" == "True" ]]; then
+      echo "Xamarin private packages NOT configured. Building a PR."
```

Comment:
If we're building on machines that has access to maccore, even for PR builds, we might as well always pass `--enable-xamarin`

Reply:
what stops someone from filing a PR that .zip's maccore and uploads it somewhere? the public build shouldn't have access to private bits if you want to trigger the pipeline automatically.

Reply:
We wouldn't trigger PRs from people without write access automatically, they'd need to be triggered manually.

Reply:
That's a bit annoying for contributors but OK :)